### PR TITLE
Move prettyprint to arrow-cast

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #.

# Rationale for this change
 

This is one of the last things left in the top-level arrow crate, the vast majority of the logic is contained within arrow-cast, and so this might as well be moved alongside it

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: arrow-flight/tests/encode_decode.rs [hunk 1]
```diff
@@ -242,9 +243,9 @@ async fn test_max_message_size_fuzz() {
         let decode_stream = FlightRecordBatchStream::new_from_flight_data(encode_stream);
         let output: Vec<_> = decode_stream.try_collect().await.expect("encode / decode");
 
-        let input_batch = concat_batches(&input[0].schema(), &input).unwrap();
-        let output_batch = concat_batches(&output[0].schema(), &output).unwrap();
-        assert_eq!(input_batch, output_batch);
+        let a = pretty_format_batches(&input).unwrap().to_string();
```
Comment (tustvold) [inline arrow-flight/tests/encode_decode.rs:246]: This avoids needing a dev-dependency on arrow-select, it is also easier to interpret the output

File: arrow-cast/src/pretty.rs [hunk 2]
```diff
@@ -266,22 +255,22 @@ mod tests {
...
+    fn test_pretty_format_dictionary() {
         // define a schema.
         let field_type =
             DataType::Dictionary(Box::new(DataType::Int32), Box::new(DataType::Utf8));
         let schema = Arc::new(Schema::new(vec![Field::new("d1", field_type, true)]));
 
         let mut builder = StringDictionaryBuilder::<Int32Type>::new();
 
-        builder.append("one")?;
+        builder.append_value("one");
```
Comment (tustvold) [inline arrow-cast/src/pretty.rs:266]: This is the infallible version, it does the same thing

File: arrow-cast/src/pretty.rs [hunk 3]
```diff
@@ -470,22 +419,6 @@ mod tests {
         assert_eq!(expected, actual, "Actual result:\n\n{actual:#?}\n\n");
     }
 
-    #[test]
```
Comment (tustvold) [inline arrow-cast/src/pretty.rs:473]: Moved to arrow/array_cast.rs integration test as chrono-tz is not a feature of arrow-cast

**Review by crepererum — APPROVED** [review]

File: arrow/Cargo.toml [hunk 4]
```diff
@@ -60,7 +60,6 @@ arrow-select = { version = "34.0.0", path = "../arrow-select" }
 arrow-string = { version = "34.0.0", path = "../arrow-string" }
 
 rand = { version = "0.8", default-features = false, features = ["std", "std_rng"], optional = true }
-comfy-table = { version = "6.0", optional = true, default-features = false }
```
Comment (alamb) [inline arrow/Cargo.toml:63]: 🎉

File: arrow/src/util/mod.rs [hunk 5]
```diff
@@ -25,7 +25,7 @@ pub mod bench_util;
 #[cfg(feature = "test_utils")]
 pub mod data_gen;
 #[cfg(feature = "prettyprint")]
-pub mod pretty;
+pub use arrow_cast::pretty;
```
Comment (alamb) [inline arrow/src/util/mod.rs:28]: 👍  for being backwards compatible

File: arrow-flight/tests/encode_decode.rs (hunk 1 above)
Comment (alamb) [reply arrow-flight/tests/encode_decode.rs:246]: Can we also please add an assert that `input.schema() == `output.schema()` ?

**Review by alamb — APPROVED** [review]
Thanks @tustvold

**Pull request merged**